gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/FlowTriggerHandler.java

```diff
@@ -62,15 +63,15 @@
 public class FlowTriggerHandler {
   private final int schedulerMaxBackoffMillis;
   private static Random random = new Random();
-  protected MultiActiveLeaseArbiter multiActiveLeaseArbiter;
+  protected Optional<MultiActiveLeaseArbiter> multiActiveLeaseArbiter;
   protected SchedulerService schedulerService;
-  protected DagActionStore dagActionStore;
+  protected Optional<DagActionStore> dagActionStore;
```

Comment:
great!  I strongly prefer this to `null` ;)

even so, since the two must work together, the ultimate abstraction would be a Strategy interface w/ capability to persist the flow action and handle the trigger event.  in that approach, this ctor would pass these two optionals to a factory to get an appropriate instance of the strategy.  when both present, the factory would give a strategy using them both, but when they're both missing, a "null strategy" would be returned.  the factory would throw if only one were given.

Reply:
That's a good idea! I can think about adding that interface in the future. By null strategy do you mean doing nothing? I agree with throwing an error if only one is given.

Reply:
yes, "Null Pattern" means, in essence, to code the operations as a no-op